src/rules/groupedImportsRule.ts

```diff
@@ -0,0 +1,171 @@
+    CURRENT_DIRECTORY_IMPORT = 3, // starts with "./"
+}
+
+interface ImportStatement {
+    statement: ts.ImportDeclaration;
+    type: ImportStatementType;
+    lineStart: number;
+    lineEnd: number;
+}
+
+class Walker extends Lint.AbstractWalker<Lint.IOptions> {
+    private previousImportStatement: ImportStatement | undefined;
+
+    public walk(sourceFile: ts.SourceFile): void {
+        const importsStatements = sourceFile.statements
+            .filter(isImportDeclaration)
+            .map(this.toImportStatement);
+        const firstFailure = importsStatements.find(this.isBadlyPositioned);
+        if (firstFailure != null) {
+            const fix = this.createFix(importsStatements);
+            this.addFailureAtNode(firstFailure.statement, Rule.GROUPED_IMPORTS, fix);
+        }
+    }
+
+    private isBadlyPositioned = (importStatement: ImportStatement): boolean => {
+        if (this.previousImportStatement != null) {
+            if (importStatement.type === this.previousImportStatement.type) {
+                if (importStatement.lineStart !== this.previousImportStatement.lineEnd + 1) {
+                    return true;
+                }
+            } else {
+                if (importStatement.type < this.previousImportStatement.type) {
+                    return true;
+                } else if (importStatement.lineStart !== this.previousImportStatement.lineEnd + 2) {
+                    return true;
+                }
+            }
+        }
+        this.previousImportStatement = importStatement;
+        return false;
+    }
+
+    private toImportStatement = (statement: ts.ImportDeclaration): ImportStatement => {
+        return {
+            lineEnd: this.sourceFile.getLineAndCharacterOfPosition(statement.getEnd()).line,
+            lineStart: this.sourceFile.getLineAndCharacterOfPosition(statement.getStart()).line,
+            statement,
+            type: this.getImportStatementType(statement),
+        };
+    }
+
+    private getImportStatementType(statement: ts.ImportDeclaration): ImportStatementType {
+        const path = this.getImportPath(statement);
+        if (path.charAt(0) === ".") {
+            if (path.charAt(1) === ".") {
+                return ImportStatementType.PARENT_DIRECTORY_IMPORT;
+            } else {
+                return ImportStatementType.CURRENT_DIRECTORY_IMPORT;
+            }
+        } else {
+            return ImportStatementType.LIBRARY_IMPORT;
+        }
+    }
+
+    private getImportPath(statement: ts.ImportDeclaration): string {
+        if (isTextualLiteral(statement.moduleSpecifier)) {
+            return statement.moduleSpecifier.text;
+        }
+        return "";
+    }
+
+    private createFix(importStatements: ImportStatement[]): Lint.Fix {
+        const newLine = this.getEofChar(this.sourceFile);
+        const imports = this.getOrderedImports(importStatements);
+        const addition = Lint.Replacement.appendText(0, imports.join(newLine));
```

Comment:
Instead of `0` consider using the start of the first import found as insertion position.

If you use 0, you'll break source files that contain a shebang (e.g. `#! env node`) in the first line.